spark/src/main/scala/org/apache/mahout/sparkbindings/blas/AtB.scala

```diff
@@ -27,92 +32,330 @@ import org.apache.spark.SparkContext._
 import org.apache.log4j.Logger
 import org.apache.mahout.math.drm.logical.OpAtB
 
+import scala.collection.mutable.ArrayBuffer
+
 object AtB {
 
-  private val log = Logger.getLogger(AtB.getClass)
+  private final implicit val log = getLog(AtB.getClass)
 
+  def atb[A: ClassTag](operator: OpAtB[A], srcA: DrmRddInput[A], srcB: DrmRddInput[A]): DrmRddInput[Int] = {
+    atb_nograph_mmul(operator, srcA, srcB, operator.A.partitioningTag == operator.B.partitioningTag)
+  }
   /**
    * The logic for computing A'B is pretty much map-side generation of partial outer product blocks
    * over co-grouped rows of A and B. If A and B are identically partitioned, we can just directly
    * zip all the rows. Otherwise, we need to inner-join them first.
+   *
    */
-  def atb_nograph[A: ClassTag](
-      operator: OpAtB[A],
-      srcA: DrmRddInput[A],
-      srcB: DrmRddInput[A],
-      zippable:Boolean = false
-      ): DrmRddInput[Int] = {
+  @deprecated("slow, will remove", since = "0.10.2")
```

Comment:
it's replaced. remove is the plan. experiments show it's decisively losing
in current spark implementations. May change though with future spark
versions. one never can be sure.

On Thu, Jun 11, 2015 at 5:09 PM, Andrew Musselman <notifications@github.com>
wrote:

> In spark/src/main/scala/org/apache/mahout/sparkbindings/blas/AtB.scala
> <https://github.com/apache/mahout/pull/135#discussion_r32281497>:
>
> >     */
> > -  def atb_nograph[A: ClassTag](
> > -      operator: OpAtB[A],
> > -      srcA: DrmRddInput[A],
> > -      srcB: DrmRddInput[A],
> > -      zippable:Boolean = false
> > -      ): DrmRddInput[Int] = {
> > +  @deprecated("slow, will remove", since = "0.10.2")
>
> Do we have a plan to improve/remove?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/mahout/pull/135/files#r32281497>.
>